ghostwriter/static/js/tinymce/plugins/ghostwriter-comments/plugin.min.js

```diff
@@ -0,0 +1,110 @@
+tinymce.PluginManager.add('ghostwriter-comments', function(editor) {
+    // Function to decode the users hex username into ascii
+    function hex2ascii(hexString) {
+        let decodedStr = '';
+        for (let i = 0; i < hexString.length; i += 2) {
+            const byte = parseInt(hexString.substr(i, 2), 16);
+            decodedStr += String.fromCharCode(byte);
+        }
+        return decodedStr
+    }
+
+    // Create Comment
+    function createComment(comment, selection) {
+        const commentNode = editor.selection.getNode();
+        let commentThread = readComments(commentNode);
+        commentThread.push({ author: username, comment: comment });
+        editor.execCommand('mceInsertContent', false, `<span class="comment" comment-data="${encodeURIComponent(JSON.stringify(commentThread))}">${selection}</span>`);
+    }
+
+    // Read Comments
+    function readComments(commentNode) {
+        let commentThread = [];
+        if (commentNode.nodeName === 'SPAN' && commentNode.hasAttribute('comment-data')) {
+            commentDataDecoded = decodeURIComponent(commentNode.getAttribute('comment-data'));
+            commentData = JSON.parse(commentDataDecoded);
+            commentData.forEach(item => {
+                const author = item.author;
+                const comment = item.comment;
+                commentThread.push({ author: author, comment: comment });
+            });
+        }
+        return commentThread;
+    }
+
+    // Update Comments
+
+    // Resolve Comments
+    function resolveComment(commentNode) {
+        commentNode.className = 'comment-resolved';
+    }
+
+    // Delete Comments
+
+
+    editor.ui.registry.addMenuItem('comment', {
+        icon: 'comment',
+        text: 'Comment',
+        onAction: function () {
+            const selection = editor.selection.getContent();
+            if (selection === '') {
+                return;
+            }
+            const node = editor.selection.getNode();
```

Comment:
This should search the selected node's ancestors for comment nodes as well - otherwise, selecting or clicking on a subspan will not work properly (ex. if you comment on a whole paragraph but click on a bolded section).